pkg/api/v1/types/types.go

```diff
@@ -112,6 +113,11 @@ func (c *ConditionUpdate) MergeExisting(existing *ptypes.Condition, compareResou
 		return nil, errBadUpdateTarget
 	}
 
+	if existing.ID != c.ID {
```

Comment:
I like this. It compels the controller to track the ID of the condition it is processing, but I don't think that's onerous.

Reply:
👍